portfolio/src/main/java/com/google/sps/data/Comments.java

```diff
@@ -32,6 +35,50 @@ public Comment(Date date, String text, long rating) {
             this.text = text;
             this.rating = rating;
         }
+
+        public Date getDate() {
+            return date;
+        }
+
+        public String getText() {
+            return text;
+        }
+
+        /**
+         * Gets the comment from comment entity.
+         * @param entity    Entity of the "Comment" kind
+         * @return A Comment object with fields retrieved from the entity.
+         */
+        public static Comment getCommentFromEntity(Entity entity) {
+            Date date = (Date) entity.getProperty(DATE);
+            String text = (String) entity.getProperty(TEXT);
+            long rating = (long) entity.getProperty(RATING);
+            return new Comment(date, text, rating);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+            Comment otherComment = (Comment) obj;
+            return Objects.equals(this.date, otherComment.date)
+                    && this.rating == otherComment.rating
+                    && Objects.equals(this.text, otherComment.text);
+        }
+
+        //Idea from effective Java : Item 9
```

Comment:
You can omit that (up to you)... I think a lot of IDEs and linters suggest you adding a `hashCode` method if you add an `equals`